services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtRequestBodyStreamAdapter.java

```diff
@@ -28,14 +28,16 @@
 @SdkInternalApi
 public final class S3CrtRequestBodyStreamAdapter implements HttpRequestBodyStream {
     private static final long MINIMUM_BYTES_BUFFERED = 1024 * 1024L;
+    // for 16 kb chunks, this limits to about 16 MB (2x the standard crt provided buffer size)
```

Comment:
Where does the 16KB come from? Is it related to https://github.com/aws/aws-sdk-java-v2/pull/6542?